cpp/src/games/nim/main/UnitTests.cpp

```diff
@@ -0,0 +1,96 @@
+#include <gtest/gtest.h>
+#include <games/nim/Game.hpp>
+
+using Game = nim::Game;
+using State = Game::State;
+using StateHistory = Game::StateHistory;
+using PolicyTensor = Game::Types::PolicyTensor;
+using IO = Game::IO;
+using Rules = Game::Rules;
+using SymmetryGroup = groups::TrivialGroup;
+using GameResults = core::WinShareResults<Game::Constants::kNumPlayers>;
+
+TEST(NimGameTest, InitialState) {
+  StateHistory history;
+  history.initialize(Rules{});
+  State state = history.current();
+
+  EXPECT_EQ(Rules::get_current_player(state), 0);
+  EXPECT_EQ(state.stones_left, 21); // Assuming the game starts with 21 stones
+}
+
+TEST(NimGameTest, MakeMove) {
+  StateHistory history;
+  history.initialize(Rules{});
+  Rules::apply(history, 2);
+  State state = history.current();
+
+  EXPECT_EQ(state.stones_left, 18);
+  EXPECT_EQ(Rules::get_current_player(state), 1);
+}
+
+TEST(NimGameTest, Player0Wins) {
+  StateHistory history;
+  history.initialize(Rules{});
+  Rules::apply(history, 2); // Player 0
+  Rules::apply(history, 2); // Player 1
+  Rules::apply(history, 2); // Player 0
+  Rules::apply(history, 2); // Player 1
+  Rules::apply(history, 2); // Player 0
+  Rules::apply(history, 2); // Player 1
+  Rules::apply(history, 2); // Player 0
+
+  GameResults::Tensor outcome;
+  core::action_t last_action = 2;
```

Comment:
It's not good that you need to set `last_action` here to the same value that you have above. It's easy for someone to change one line but forget to change the other.

Better would be to have:

```
std::vector<core::action_t> actions = { kTake3, kTake3, kTake3, kTake3, kTake3, kTake3, kTake3 };

for (core::action_t action : actions) {
  Rules::apply(history, action);
}

core::action_t last_action = actions.back();
```